HeterogeneousCore/AlpakaInterface/interface/atomicInc.h

```diff
@@ -0,0 +1,15 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_atomicInc_h
+#define HeterogeneousCore_AlpakaInterface_interface_atomicInc_h
+
+#include <alpaka/alpaka.hpp>
+
+// This function is similar to atomicInc, but deduces the limiting value from the type itself.
+
+ALPAKA_NO_HOST_ACC_WARNING
+template <typename TAcc, typename T, typename THierarchy = alpaka::hierarchy::Grids>
+ALPAKA_FN_HOST_ACC auto atomicInc(TAcc const& acc, T* address, THierarchy const& hierarchy = THierarchy()) -> T {
```

Comment:
What are you thoughts on this function being defined in the global namespace vs. `cms::alpakatools`?

We have tended to place these kind of generic functions in `cms::alpakatools` namespace, but I see (now) we already broke off from that convention with `atomicMaxF.h` and `atomicMaxPair.h`.